src/ui/assets-and-images.md

```diff
@@ -211,36 +177,27 @@ That is, if `.../my_icon.png` is 72px by 72px, then
 but they both render into 72px by 72px (in logical pixels),
 if width and height are not specified.
 
-Each entry in the asset section of the `pubspec.yaml`
+{{site.alert.note}}
+  [Device pixel ratio][] depends on [MediaQueryData.size][] which requires to have either a [MaterialApp][] or [CupertinoApp][] as an ancestor of your [`AssetImage`][].
+{{site.alert.end}}
+
+#### Bundling of resolution-aware image assets {#resolution-aware-bundling}
+
+Only the main asset (or its parent directory) needs
+to specified in the assets section of `pubspec.yaml`; the
+variants will be bundled automatically. Each entry
```

Comment:
**issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
```suggestion
You only need to specify the main asset or its parent directory
in the `assets` section of `pubspec.yaml`.
Flutter bundles the variants for you.
Each entry
```